src/ast/passes/codegen_llvm.cpp

```diff
@@ -157,6 +157,17 @@ void CodegenLLVM::kstack_ustack(const std::string &ident,
   const auto uint64_size = sizeof(uint64_t);
 
   AllocaInst *stackid = b_.CreateAllocaBPF(CreateUInt64(), "stackid");
+  AllocaInst *nr_stack_frames = b_.CreateAllocaBPF(CreateUInt32(),
+                                                   "nr_stack_frames");
+
+  std::vector<llvm::Type *> stack_key_elements = {
+    b_.getInt64Ty(), // stack id
+    b_.getInt32Ty(), // nr_stack_frames
+  };
+  StructType *stack_key_struct = b_.GetStructType("stack_key",
+                                                  stack_key_elements,
+                                                  false);
+  AllocaInst *stack_key = b_.CreateAllocaBPF(stack_key_struct, "stack_key");
```

Comment:
Is there a reason to store `stackid` and `nr_stack_frames` both as scalars and inside the `stack_key` struct? I know that LLVM will probably optimize this away but it just seems like a waste of stack space (and is a bit difficult to read).

Reply:
I think I originally did it because ustack returns a different struck that includes the pid and probe id but I could just use the stack_key and GEP to pull out the `stackid` and `nr_stack_frames` for this case. Let me try that and see if it makes it more readable.

Reply:
Ok, that was easy enough to change.